docs/contributor/testing.md

```diff
@@ -33,6 +34,17 @@ Unit and Env tests follow the [Go convention](https://go.dev/doc/tutorial/add-a-
 | Unit (located with the individual source files) | Unit | It tests the units in isolation. This test suite assesses the implementation correctness of the units of business logic. |
 | Env-tests (located with the individual source files) | Integration  (low-level) | It tests the behaviour of the Telemetry Manager in integration with a Kubernetes API server replaced with a test double. This test suite assesses the integration correctness of the Telemetry Manager. |
 | [E2E](/test/e2e) | Acceptance / Integration (high-level) | It tests the usability scenarios of the Telemetry Manager in a cluster. This test suite assesses the functional correctness of the Telemetry Manager. |
+| Operational | System | It validates the operational aspects of the module (successful module upgrades, etc.). |
+​
+#### Testing of new functionality
+​
+Testing a new functionality encompasses two activities: capturing the acceptance criteria for each new functionality and providing the requirements traceability. The `Requirement` ↔︎ `Test Suite` linkage is achieved by the Behavioural-Driven Testing DSL of the [Ginkgo](https://onsi.github.io/ginkgo/) testing framework. The `User Story` ↔︎ `Acceptance Criteria` linkage is provided semi-automatic by making acceptance criteria a mandatory part of each story with new functionality (using the PR-template checklist).
```

Comment:
```suggestion
Testing a new functionality encompasses two activities: capturing the acceptance criteria for each new functionality and providing the requirements traceability. `Requirement` and `Test Suite` are linked by the Behavioural-Driven Testing DSL of the [Ginkgo](https://onsi.github.io/ginkgo/) testing framework. `User Story` and `Acceptance Criteria` are linked semi-automatically because acceptance criteria are a mandatory part of each user story with new functionality (using the PR-template checklist).
```

Why do you love nouns so much?? :D

Reply:
I reckon that is the way I experience the world around me :) I blame object-oriented analysis and domain-driven design for this.